topdown/http.go

```diff
@@ -936,15 +933,22 @@ func getResponseHeaderDate(headers http.Header) (date time.Time, err error) {
 		err = fmt.Errorf("no date header")
 		return
 	}
-	return time.Parse(time.RFC1123, dateHeader)
+	return http.ParseTime(dateHeader)
 }
 
-func getResponseHeaderExpires(headers http.Header) (date time.Time, err error) {
+func getResponseHeaderExpires(headers http.Header) time.Time {
 	expiresHeader := headers.Get("expires")
 	if expiresHeader == "" {
-		return
+		return time.Time{}
 	}
-	return time.Parse(time.RFC1123, expiresHeader)
+
+	date, err := http.ParseTime(expiresHeader)
+	if err != nil {
+		// servers can set `Expires: 0` which is an invalid date to indicate expired content
```

Comment:
I suppose we could have checked explicitly if the expires header is "0" instead? The way it is done now, we could ignore errors when it's neither valid nor "0".... 🤔 

(Just checked, that's what @anderseknert has proposed, too... 😄)

Reply:
The [spec](https://tools.ietf.org/html/rfc7234#section-5.3) says:
> A cache recipient MUST interpret invalid date formats, especially the value "0", as representing a time in the past (i.e., "already expired").

So while it does call out "0" especially, it seems fine to treat any other invalid dates the same way.

What I proposed was rather that we don't check these headers _at all_ when the client is configured with `force_cache: true` + `force_cache_duration` 😊

Reply:
Ah I see. Thanks for digging out the spec! 🔍 

Sounds like we should get this merged then :D